packages/@lwc/ssr-compiler/src/compile-template/transformers/element.ts

```diff
@@ -138,6 +143,11 @@ function yieldAttrOrPropDynamicValue(
     cxt.import('htmlEscape');
     const isHtmlBooleanAttr = isBooleanAttribute(name, elementName);
     const scopedExpression = getScopedExpression(value as EsExpression, cxt);
+
+    if (name === 'class') {
+        cxt.import('normalizeClass');
+    }
+
     return [bYieldDynamicValue(b.literal(name), scopedExpression, b.literal(isHtmlBooleanAttr))];
```

Comment:
I'm thinking `bYieldDynamicValue` should be broken down for class and other attributes, even though treeshaking will take care of not including normalizeClass. There are other quirks that might be easier to debug and fix.